File: C++/DSA/Problems/Leetcode/LC1.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <unordered_map>
using namespace std;

vector<int> twoSum(vector<int>& nums, int target) {
    unordered_map<int, int> seen;
    for (int i = 0; i < nums.size(); i++) {
        int complement = target - nums[i];
        if (seen.find(complement) != seen.end()) {
            return {seen[complement], i};
        }
        seen[nums[i]] = i;
    }
    return {};
}


int main(){
    vector<int> nums = {3,4,6};
    int target = 10;
    
    vector<int> res = twoSum(nums, target);
    cout << res[0] << " " << res[1] << endl;
}

File: Python/DSA/Problems/Leetcode/LC1.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        elements = dict()
        for i, item in enumerate(nums):
            if target - item in elements:
                return [i, elements[target - item]]
            else:
                elements[item] = i
        return [-1, -1]


def main():
    nums = [2, 7, 11, 15]
    target = 9

    sol = Solution()
    result = sol.twoSum(nums, target)

    print("Nums   :", nums)
    print("Target :", target)
    print("Indices:", result)
    print("Values :", [nums[result[0]], nums[result[1]]])


if __name__ == "__main__":
    main()